src/Illuminate/Broadcasting/BroadcastManager.php

```diff
@@ -74,14 +74,42 @@ public function routes(array $attributes = null)
                 ['get', 'post'], '/broadcasting/auth',
                 '\\'.BroadcastController::class.'@authenticate'
             )->withoutMiddleware([\Illuminate\Foundation\Http\Middleware\VerifyCsrfToken::class]);
+        });
+    }
+
+    /**
+     * Register the routes for handling broadcast authentication.
+     *
+     * @param  array|null  $attributes
+     * @return void
+     */
+    public function userAuthenticationRoutes(array $attributes = null)
+    {
+        if ($this->app instanceof CachesRoutes && $this->app->routesAreCached()) {
+            return;
+        }
 
+        $attributes = $attributes ?: ['middleware' => ['web']];
+
+        $this->app['router']->group($attributes, function ($router) {
             $router->match(
                 ['get', 'post'], '/broadcasting/user-auth',
                 '\\'.BroadcastController::class.'@authenticateUser'
             )->withoutMiddleware([\Illuminate\Foundation\Http\Middleware\VerifyCsrfToken::class]);
         });
     }
 
+    /**
+     * Alias for routes().
+     *
+     * @param  array|null  $attributes
+     * @return void
+     */
+    public function channelAuthorizationRoutes(array $attributes = null)
```

Comment:
Probably best to just make this an empty array fallback rather than use two different types.